airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -414,7 +414,14 @@ def write_dag(
         serialized_dag_hash = session.scalars(
             select(cls.dag_hash).where(cls.dag_id == dag.dag_id).order_by(cls.created_at.desc())
         ).first()
-        dag_version = DagVersion.get_latest_version(dag.dag_id, session=session)
+        dag_version = session.scalar(
+            select(DagVersion)
+            .where(DagVersion.dag_id == dag.dag_id)
+            .options(joinedload(DagVersion.task_instances))
+            .options(joinedload(DagVersion.serialized_dag))
```

Comment:
This is wrong -- why are we loading the entire serialized dag again before checking if the hash has changed or not.

cc @ephraimbuddy